### PR TITLE
[oatpp-zlib] add missing deps

### PR DESCRIPTION
**Pull request by autoantwort**

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
usage:
```
oatpp-zlib provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(oatpp-zlib CONFIG REQUIRED)
    target_link_libraries(main PRIVATE oatpp::oatpp-zlib)
```

When testing usage, the following error occurs:
```
1> [CMake] CMake Error at G:/oatpp-zlib/installed/x64-windows/share/oatpp-zlib/oatpp-zlibTargets.cmake:60 (set_target_properties):
1> [CMake]   The link interface of target "oatpp::oatpp-zlib" contains:
1> [CMake] 
1> [CMake]     oatpp::oatpp
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   G:/oatpp-zlib/installed/x64-windows/share/oatpp-zlib/oatpp-zlibConfig.cmake:18 (include)
1> [CMake]   G:/oatpp-zlib/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   CMakeUsage/CMakeLists.txt:13 (find_package)
```
```
1> [CMake] CMake Error at G:/oatpp-zli/installed/x64-windows/share/oatpp-zlib/oatpp-zlibTargets.cmake:60 (set_target_properties):
1> [CMake]   The link interface of target "oatpp::oatpp-zlib" contains:
1> [CMake] 
1> [CMake]     ZLIB::ZLIB
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   G:/oatpp-zli/installed/x64-windows/share/oatpp-zlib/oatpp-zlibConfig.cmake:21 (include)
1> [CMake]   G:/oatpp-zli/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   CMakeUsage/CMakeLists.txt:13 (find_package)
```

<details>
<summary><b>CMakeFindUsage.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "oatpp-zlib/Processor.hpp"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "E:/oatpp-zlib/scripts/buildsystems/vcpkg.cmake")

project ("CMakeFindUsage")

add_library (main "CMakeFindUsage.cpp")

find_package(oatpp-zlib CONFIG REQUIRED)
target_link_libraries(main PRIVATE oatpp::oatpp-zlib)
</pre></details>

**Comment by autoantwort** [comment]
Does it works now?

File: ports/oatpp-zlib/missing-find_dependency.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/cmake/module-config.cmake.in b/cmake/module-config.cmake.in
+index 5cc12b0..71f8f10 100644
+--- a/cmake/module-config.cmake.in
++++ b/cmake/module-config.cmake.in
+@@ -1,5 +1,8 @@
+ @PACKAGE_INIT@
+ 
++find_dependency(oatpp)
++find_dependency(ZLIB)
```
Comment (MonicaLiu0311) [inline ports/oatpp-zlib/missing-find_dependency.patch:9]:
```suggestion
+include(CMakeFindDependencyMacro)
+find_dependency(oatpp)
+find_dependency(ZLIB)
```

**Comment by MonicaLiu0311** [comment]
> Does it works now?

There are still errors, the test code is the same as above: https://github.com/microsoft/vcpkg/pull/33605#issuecomment-1709417671

```
>------ Build All started: Project: CMakeUsage, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/monica/source/repos/CMakeUsage/CMakeUsage/CMakeLists.txt
    CMakeUsage.cpp
G:\oatpp-zlib\installed\x64-windows\include\oatpp-1.3.0\oatpp-zlib\oatpp-zlib\Processor.hpp(28,10): fatal error C1083: Cannot open include file: 'oatpp/core/data/buffer/Processor.hpp': No such file or directory 

Build All failed.
```

**Comment by autoantwort** [comment]
Locally the usage is now working :)

**Comment by MonicaLiu0311** [comment]
oatapp-zlib
-
The usage test passed on `x64-windows` (header files found):
```
oatpp-zlib provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(oatpp-zlib CONFIG REQUIRED)
    target_link_libraries(main PRIVATE oatpp::oatpp-zlib)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**